src/pybind/mgr/dashboard/controllers/oauth2.py

```diff
@@ -0,0 +1,29 @@
+import cherrypy
+
+from dashboard.controllers._auth import ControllerAuthMixin
+from dashboard.exceptions import DashboardException
+from dashboard.services.auth.oauth2 import OAuth2
+
+from .. import mgr
+from . import BaseController, Endpoint, Router
+
+
+@Router('/auth/oauth2', secure=False)
+class Oauth2(BaseController, ControllerAuthMixin):
+
+    @Endpoint(json_response=False, version=None)
+    def login(self):
+        if not mgr.get_module_option('sso_oauth2'):
+            raise DashboardException(500, msg='Failed to login: SSO OAuth2 is not enabled')
+
+        token = OAuth2.get_token(cherrypy.request)
+        if not token:
+            raise cherrypy.HTTPError(400, message='Failed to login: Missing access token')
+
+        raise cherrypy.HTTPRedirect(OAuth2.get_login_redirect_url(token))
+
+    @Endpoint(json_response=False, version=None)
+    def logout(self):
+        token = OAuth2.get_token(cherrypy.request)
```

Comment:
For consistency, shouldn't you also check if OAuth2 is enabled here?

Reply:
Makes sense, added the check